report-api/src/main/java/no/sikt/nva/data/report/api/fetch/model/ReportRequest.java

```diff
@@ -124,4 +112,16 @@ private static String extractReportType(RequestInfo requestInfo) {
     private static String extractFormat(RequestInfo requestInfo) {
         return requestInfo.getHeaders().getOrDefault(ACCEPT, null);
     }
+
+    private void validate() throws BadRequestException {
+        if (this.offset < 0) {
+            throw new BadRequestException("Offset cannot be less than zero");
+        }
+        if (this.pageSize < 0) {
+            throw new BadRequestException("Page size cannot be less than zero");
+        }
+        if (this.after.isAfter(this.before)) {
+            throw new BadRequestException("Logically, 'before should be after the value of 'after'");
+        }
```

Comment:
Is it correct? Shouldn't it be like this?

```
if (this.after.isBefore(this.before)) {
    throw new BadRequestException("'after' should not be before 'before'");
}
```

Reply:
https://github.com/BIBSYSDEV/nva-data-report-api/pull/2#discussion_r1407320364 😄 Can change this, but in separate PR

Reply:
I would also change error message to not repeat "..be before 'before'".
`Usually, in time comparisons 'after' is a moment later in time than 'before'.`